src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -727,10 +727,34 @@ namespace cryptonote
   int t_cryptonote_protocol_handler<t_core>::handle_uptime_proof(int command, NOTIFY_UPTIME_PROOF::request& arg, cryptonote_connection_context& context)
   {
     MLOG_P2P_MESSAGE("Received NOTIFY_UPTIME_PROOF");
-    if(context.m_state != cryptonote_connection_context::state_normal)
+    if(context.m_state <= cryptonote_connection_context::state_synchronizing)
```

Comment:
Is this early return really necessary?  It kind of seems like we should accept the proof regardless of the peer state.

Reply:
Hmm. Its only here because it mirrors with: https://github.com/loki-project/loki/blob/e6520522b0ca8462e5b87e022817386afcd01911/src/cryptonote_protocol/cryptonote_protocol_handler.h#L141

I don't really have an opinion on whether or not to keep, rather just for consistency.

Reply:
I think we should remove it because (assuming I'm reading this right -- this is for incoming requests, right?) we generally want to follow the principle of "conservative in what you send, liberal in what you accept."  In other words, if they sent it even if we think they shouldn't have we should probably just accept it.  (But also a little because "we think they shouldn't have" could be wrong -- i.e. in a race condition with some other status change).

Reply:
I'm don't think their state can change in the context object once the request has been made. Accepting it would decrease the uptime failure cases so, yes lets be liberal in accepting